components/input/textarea-count.component.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+import { AfterContentInit, ChangeDetectionStrategy, Component, ContentChild, ElementRef, Input, OnDestroy, Renderer2 } from '@angular/core';
+import { isNotNil } from 'ng-zorro-antd/core/util';
+import { NzInputDirective } from 'ng-zorro-antd/input/input.directive';
+import { EMPTY, Subject } from 'rxjs';
+import { map, startWith, takeUntil } from 'rxjs/operators';
+
+@Component({
+  selector: 'nz-textarea-count',
+  template: `
+    <ng-content select="textarea[nz-input]"></ng-content>
+  `,
+  host: {
+    class: 'ant-input-textarea-show-count'
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class NzTextareaCountComponent implements AfterContentInit, OnDestroy {
+  @ContentChild(NzInputDirective, { static: true }) nzInputDirective!: NzInputDirective;
+  @Input() nzMaxCharacterCount: number = 0;
+  @Input() nzComputeCharacterCount: (v: string) => number = v => v.length;
+
+  private destroy$ = new Subject();
+
+  constructor(private renderer: Renderer2, private elementRef: ElementRef<HTMLElement>) {}
+  ngAfterContentInit(): void {
+    if (this.nzInputDirective.ngControl) {
+      const valueChanges = this.nzInputDirective.ngControl.valueChanges || EMPTY;
+      valueChanges
+        .pipe(
+          takeUntil(this.destroy$),
+          map(() => this.nzInputDirective.ngControl.value),
+          startWith(this.nzInputDirective.ngControl.value as string)
+        )
+        .subscribe(value => {
+          this.onValueChange(value);
+        });
+    }
+  }
+
+  onValueChange(value: unknown): void {
```

Comment:
I don't think this needs to be unknown, the value of a textarea is always a string (and above there's also a type assertion for it, too). Also makes `String(value)` superfluous.

Reply:
The value of a textarea could be an emoji, like 😄

Reply:
That'd still be a string.